src/Arcus.Security.Tests.Unit/Core/CachedSecretProviderTests.cs

```diff
@@ -72,171 +72,265 @@ public void CachedSecretProvider_CreateWithCorrectArguments_ShouldSucceed()
             Assert.NotNull(cachedSecretProvider);
         }
 
-        [Theory]
-        [InlineData(GetCachedSecret.WithinCacheInterval)]
-        [InlineData(GetCachedSecret.OutsideCacheInterval)]
-        [InlineData(GetCachedSecret.SkippedCache)]
-        public async Task CachedSecretProvider_GetTwoSecretValues_WithCaching(GetCachedSecret cacheSetup)
+        [Fact]
+        public async Task CachedSecretProvider_GetTwoSecretValues_WithinCacheInterval_ShouldReturnTheSameValues()
         {
             // Arrange
             string expectedFirstSecret = Guid.NewGuid().ToString("N");
             string expectedSecondSecret = Guid.NewGuid().ToString("N");
 
-            // Act 
-            (string actualFirstSecret, string actualSecondSecret) = await GetTwoSecretsFromCachedProvider(
-                (expectedFirstSecret, expectedSecondSecret),
-                (provider, keyName) => provider.Get(keyName),
-                cacheSetup);
+            var spyTestProvider = new TestSecretProviderStub(expectedFirstSecret);
+            TimeSpan cacheInterval = TimeSpan.FromSeconds(3);
+
+            const string keyName = "MyValue";
+            ICachedSecretProvider cachedSecretProvider = new CachedSecretProvider(
+                spyTestProvider,
+                new CacheConfiguration(cacheInterval),
+                new MemoryCache(new MemoryCacheOptions()));
+
+            // Act
+            string actualFirst = await cachedSecretProvider.Get(keyName);
+            await Task.Delay(TimeSpan.Zero);
```

Comment:
Given timespan is zero, do we actually need it?

Reply:
Anyhow, merged it anyway. Feel free to create a new PR if you agree.

Reply:
Ah, could be changed to 100ms or so.